superset-frontend/src/views/CRUD/alert/components/AlertReportCronScheduler.tsx

```diff
@@ -16,44 +16,59 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import React, { useState, useCallback, useRef, FunctionComponent } from 'react';
+import React, { useState, useCallback, useRef, FocusEvent } from 'react';
 import { t, useTheme } from '@superset-ui/core';
 
-import { AntdInput } from 'src/components';
+import { AntdInput, RadioChangeEvent } from 'src/components';
 import { Input } from 'src/components/Input';
 import { Radio } from 'src/components/Radio';
 import { CronPicker, CronError } from 'src/components/CronPicker';
 import { StyledInputContainer } from 'src/views/CRUD/alert/AlertReportModal';
 
-interface AlertReportCronSchedulerProps {
+export interface AlertReportCronSchedulerProps {
   value: string;
   onChange: (change: string) => any;
 }
 
-export const AlertReportCronScheduler: FunctionComponent<AlertReportCronSchedulerProps> =
+export const AlertReportCronScheduler: React.FC<AlertReportCronSchedulerProps> =
   ({ value, onChange }) => {
     const theme = useTheme();
     const inputRef = useRef<AntdInput>(null);
     const [scheduleFormat, setScheduleFormat] = useState<'picker' | 'input'>(
       'picker',
     );
+
+    const handleRadioButtonChange = useCallback(
+      (e: RadioChangeEvent) => setScheduleFormat(e.target.value),
+      [],
+    );
+
     const customSetValue = useCallback(
       (newValue: string) => {
         onChange(newValue);
         inputRef.current?.setValue(newValue);
       },
       [inputRef, onChange],
     );
+
+    const handleBlur = useCallback(
+      (event: FocusEvent<HTMLInputElement>) => {
+        onChange(event.target.value);
+      },
+      [onChange],
+    );
+
+    const handlePressEnter = useCallback(() => {
+      onChange(inputRef.current?.input.value || '');
+    }, [onChange]);
```

Comment:
@diegomedina248 do you think we need `useCallback` for simple Input component props?

Reply:
It doesn't hurt nor improve in this case, but I believe it's cleaner

Reply:
It's a bit more code and a bit more computation imo, which is my only concern with not much performance impact, but it's not a big deal. I'll cc @rusackas if he has any opinions.

Reply:
but to your comment of it being cleaner, @diegomedina248 I meant to also ask in what way?